rust/src/builders/tx_builder.rs

```diff
@@ -600,29 +600,31 @@ impl TransactionBuilder {
     }
 
     pub fn add_mint(&mut self, result: &MintBuilderResult) {
+        let mut mint = self.get_mint().unwrap_or_else(Mint::new);
+        let assets = {
+            let mut old_assets = mint.get(&result.policy_id).unwrap_or_else(MintAssets::new);
+            let mut new_assets = result.assets.clone();
+            old_assets.0.append(&mut new_assets.0);
+            old_assets
+        };
+        mint.insert(&result.policy_id, &assets);
+        self.mint = Some(mint);
         if let Some(ref data) = result.aggregate_witness {
             self.witness_set_builder.add_input_aggregate_witness_data(data);
             if let InputAggregateWitnessData::PlutusScript(witness, _, _) = data {
                 let redeemer = {
                     let untagged = witness.untagged_redeemer();
                     let data = untagged.datum();
                     let ex_units = untagged.ex_units();
-                    let index = self.mint.as_ref().and(Some(1u64)).or(Some(0)).unwrap();
+                    let index = self.mint.as_ref()
```

Comment:
Note to self: look into whether or not this is correct since there is still no comment here explaining whether or not this is really how mint indices are supposed to work

Reply:
Minting indices are the total length of minting assets. It cannot work like others when minting results are added in a batch. But the indices should be identical.

Reply:
Or wrap Mint in vector in transaction builder?

Reply:
Ok, this is incorrect. I think I've figure it out. Will change it soon.

Reply:
Fixed and added some unit tests.